File: solutions/fable-1/fable/python/primes.py
import sys


def main():
    data = sys.stdin.read().split()
    n = int(data[0]) if data else 0
    if n < 2:
        print()
        return
    sieve = bytearray([1]) * (n + 1)
    sieve[0] = sieve[1] = 0
    i = 2
    while i * i <= n:
        if sieve[i]:
            sieve[i * i::i] = bytearray(len(range(i * i, n + 1, i)))
        i += 1
    print(" ".join(str(i) for i in range(2, n + 1) if sieve[i]))


if __name__ == "__main__":
    main()

File: solutions/fable-1/fable/typescript/primes.ts
const input = await Bun.stdin.text();
const n = parseInt(input.trim(), 10) || 0;

const sieve = new Uint8Array(n + 1);
const primes: number[] = [];
for (let i = 2; i <= n; i++) {
  if (sieve[i] === 0) {
    primes.push(i);
    for (let j = i * i; j <= n; j += i) sieve[j] = 1;
  }
}

process.stdout.write(primes.join(" ") + "\n");
